File: javascript/storage.js
const fs = require('fs'); //Import file system module
const Task = require('./task'); //Import the Task class from task.js

const FILE_NAME = 'tasks.json';

function loadTasks() {
    try {
      const data = fs.readFileSync(FILE_NAME, 'utf-8');
      return unserializeTasks(JSON.parse(data));
    } catch (error) {
      if (error.code === 'ENOENT') {
        // File not found, return empty array
        return [];
      } else {
        throw error;
      }
    }
  }
  
function saveTasks(tasks) {
    const data = JSON.stringify(serializeTasks(tasks));
    fs.writeFileSync(FILE_NAME, data, 'utf-8');
}

function serializeTasks(tasks) {
  return tasks.map(task => task.toDict()); //Lambda function
}

function unserializeTasks(taskDicts) {
  return taskDicts.map(taskDict => Task.fromDict(taskDict));
}

module.exports = { loadTasks, saveTasks };

File: python/storage.py
import json
from task import Task

def serialize_tasks(tasks):
    return [i.to_dict() for i in tasks]

def unserialize_tasks(tasks):
    return [Task.from_dict(i) for i in tasks]

def load_tasks(task_file):
    """
    Load tasks from the file
    """
    try:
        with open(task_file, 'r') as f:
            return unserialize_tasks(json.load(f))
    except FileNotFoundError:
        return []

def save_tasks(tasks, task_file):
    """
    Save tasks to the file
    """
    with open(task_file, 'w') as f:
        json.dump(serialize_tasks(tasks), f)